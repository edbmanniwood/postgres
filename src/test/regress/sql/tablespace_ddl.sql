-- allow tablespaces with empty location so that we can
-- 1) test LOCATION, but
-- 2) not have to create a directory on the filesytem to match LOCATION
SET allow_in_place_tablespaces = true;

-- create a tablespace using WITH clause
CREATE TABLESPACE regress_tblspacewith LOCATION '' WITH (random_page_cost = 3.0); -- ok

-- see that the tablespace ddl is correctly returned
SELECT pg_get_tablespace_ddl('regress_tblspacewith');

-- drop the tablespace
DROP TABLESPACE regress_tblspacewith;

