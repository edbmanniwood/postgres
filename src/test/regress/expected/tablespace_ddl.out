-- allow tablespaces with empty location so that we can
-- 1) test LOCATION, but
-- 2) not have to create a directory on the filesytem to match LOCATION
SET allow_in_place_tablespaces = true;
-- create a tablespace using WITH clause
CREATE TABLESPACE regress_tblspacewith LOCATION '' WITH (random_page_cost = 3.0); -- ok
-- see that the tablespace ddl is correctly returned
SELECT pg_get_tablespace_ddl('regress_tblspacewith');
                              pg_get_tablespace_ddl                              
---------------------------------------------------------------------------------
 CREATE TABLESPACE regress_tblspacewith LOCATION '' WITH (random_page_cost = 3);
(1 row)

-- create tablespace owned by a particular user
CREATE USER regress_user;
CREATE TABLESPACE regress_tblspcuser OWNER regress_user LOCATION '' WITH (random_page_cost = 3.0); -- ok
-- see that the tablespace ddl is correctly returned
SELECT pg_get_tablespace_ddl('regress_tblspcuser');
                                      pg_get_tablespace_ddl                                       
--------------------------------------------------------------------------------------------------
 CREATE TABLESPACE regress_tblspcuser OWNER regress_user LOCATION '' WITH (random_page_cost = 3);
(1 row)

-- drop the tablespace
DROP TABLESPACE regress_tblspacewith;
DROP TABLESPACE regress_tblspcuser;
DROP USER regress_user;
